### PR TITLE
Allow custom log.Logger

### PR DESCRIPTION
**Pull request by jkroepke**
This PR allow to use custom log.Logger together with the promlog wrapper mechanics and can be considered as alternative to copy, own and manually patch as needed.

A potential use case is using promlog with an Windows Event Log backend. 

* https://github.com/go-kit/log/pull/31

---

Not sure, if you are willing to include Windows EventLog support by default.

### CODE REVIEW

File: promlog/log.go [hunk 1]
```diff
@@ -111,13 +111,16 @@ type Config struct {
 // New returns a new leveled oklog logger. Each logged line will be annotated
 // with a timestamp. The output always goes to stderr.
 func New(config *Config) log.Logger {
-	var l log.Logger
 	if config.Format != nil && config.Format.s == "json" {
-		l = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
+		return NewWithLogger(log.NewJSONLogger(os.Stderr), config)
 	} else {
```
Comment (obalunenko) [inline promlog/log.go:116]: Unnecessary else here - return used in if statement.

File: promlog/log.go [hunk 2]
```diff
@@ -111,13 +111,16 @@ type Config struct {
 // New returns a new leveled oklog logger. Each logged line will be annotated
 // with a timestamp. The output always goes to stderr.
 func New(config *Config) log.Logger {
-	var l log.Logger
 	if config.Format != nil && config.Format.s == "json" {
-		l = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
```
Comment (obalunenko) [inline promlog/log.go:116]: Why is it better not to use `WithSyncWriter`?

Comment (jkroepke) [reply promlog/log.go:116]: This changes is out of scope of this PR. I revert this. Sorry for the confusion here.

**Review by obalunenko — APPROVED** [review]
LGTM

**Comment by jkroepke** [comment]
@obalunenko Can I help here to move forward here? How many approvals are need to a merge?

**Comment by obalunenko** [comment]
I suppose you have to ask @kakkoyun or @roidelapluie for a review and merge - as they're maintainers of this repo

**Comment by jkroepke** [comment]
@kakkoyun or @roidelapluie: could you please take a look here?

**Comment by SuperQ** [comment]
lolol, github outage duplicated my LGTM.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**